### PR TITLE
Add some further stubs for syntax sections

### PR DESCRIPTION
**Pull request by awwaiid**
I'm starting to add some things to Language/syntax. I'm doing a PR here because I'm at a pause point, though these are really just stubs for a lot of further work.

### CODE REVIEW

File: doc/Language/syntax.pod [hunk 1]
```diff
@@ -341,22 +342,67 @@ arrays themselves:
...
+    my $pattern = /foo/;
+    say "Match!" if "asdfoosdf" ~~ $pattern; # Match!
+
 =head2 Declarations
 
 =comment TODO
 
-=head2 Subroutine calls
+=head3 Variable declaration
+
```
Comment (LLFourn) [inline doc/Language/syntax.pod]: Perhaps a link to http://docs.perl6.org/language/variables#Variable_Declarators_and_Scope for more info.

File: doc/Language/syntax.pod [hunk 2]
```diff
@@ -341,22 +342,67 @@ arrays themselves:
...
+    my $x;                      # simple lexical variable
+    my $x = 7;                  # initialize the variable
+    my Int $x = 7;              # declare the type
+    my Int $x:D = 7;            # specify that the value must be defined (not undef)
+    my Int $x where * > 3 = 7;  # restrict the value based on a function
```
Comment (LLFourn) [inline doc/Language/syntax.pod]: s/restrict/constrain/ (just to be consistent with the .constraints method). Maybe s/function/block/.

**Comment by LLFourn** [comment]
Looks good. I made some comments that you can take or leave :).

Thanks!

**Comment by awwaiid** [comment]
Dang. OK, I take the quick merges of my stuff to indicate that I'm doing more good than harm, so I'll keep at it! :)

**Comment by LLFourn** [comment]
Yes please!

They are doing no harm at all! I just GH commenting interface mostly. Also to reassure you that someone is actually reading what your contributing.

**Pull request merged**